### DISCUSSION TITLE
KIP-320 support

### DISCUSSION THREAD

**Discussion by scanterog** [edenhill/librdkafka#3644] [General]
Hello, 

We've recently hit the case 4 of out of range handling described [here](https://cwiki.apache.org/confluence/display/KAFKA/KIP-392%3A+Allow+consumers+to+fetch+from+closest+replica). This should be handled by [KIP-320](https://cwiki.apache.org/confluence/display/KAFKA/KIP-320%3A+Allow+fetchers+to+detect+and+handle+log+truncation) based on that doc but this is [not currently supported in librdkafka](https://github.com/edenhill/librdkafka/blob/master/INTRODUCTION.md#supported-kips).

is there any ETA to implement this or what would be the recommended way to handle this with rdkafka?

**Comment by edenhill** [comment]
Hi, KIP-320 is on our backlog for next year, but it does not have a definite prioritization yet.

**Reply by scanterog** [reply]
Thanks for the quick reply. Does it mean that for now avoiding fetch-from-replicas is probably the best way to go with rdkafka? Not being able to handle out of range errs will make the consumer to fallback on the `auto.offset.reset` property which means either data re-processing (`earliest`) or small data loss (`latest`).

An alternative would be to use `error` but IDK if the error is exposed/bubble up or just logged. I need to check that.
